Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/AttachmentReadback.h

```diff
@@ -43,14 +46,37 @@ namespace AZ
             };
 
             AttachmentReadback(const RHI::ScopeId& scopeId);
-
             virtual ~AttachmentReadback();
 
-            //! Readback a pass attachment.
+            //! Reads back a single pass attachment.
             //! @param readbackName is the name for the readback buffer. And it will be saved in ReadbackResult::m_name. If the name is empty, a name will be generated automatically.
             //! Return true if the pass attachment readback request was submitted.  
             //! The callback function set by SetCallback(CallbackFunction callback) will be called once the readback is finished
-            bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName);
+            //! @param imageViewDescriptor If null, by default it is assumed that @attachment refers to Mip Level 0.
+            //!        When different than null, the image view descriptor can be used to specify which mip level is the attachment bound to.
+            bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName, const RHI::ImageViewDescriptor* imageViewDescriptor = nullptr);
+
+
+            struct ReadbackRequestInfo
+            {
+                // Same semantic as ReadPassAttachment Arg 0.
+                const PassAttachment* m_attachment = nullptr;
+                // Same semantic as ReadPassAttachment Arg 1.
+                Name m_readbackName;
+                // With a properly set imageViewDescriptor we can readback
+                // a particular mip level.
+                RHI::ImageViewDescriptor m_imageViewDescriptor;
+            };
+            //! Reads back multiple pass attachments in one frame.
+            bool ReadPassAttachments(const AZStd::vector<ReadbackRequestInfo>& readbackAttachmentRequests);
```

Comment:
Do you intent to support a list of ReadbackRequestInfo which may have different PassAttachment?

Reply:
Yes.